app/services/hyrax/fixity_status_service.rb

```diff
@@ -0,0 +1,86 @@
+module Hyrax
+  # Creates fixity status messages to display to user, for a fileset, or
+  # a specific version/file within it. Determines status by looking up
+  # existing ChecksumAuditLog objects, does not actually do a check itself.
+  # See FileSetFixityCheckService and ChecksumAuditLog for actually performing
+  # checks and recording as ChecksumAuditLog objects.
+  class FixityStatusService
+    include ActionView::Helpers::TagHelper
```

Comment:
I don't think of services as rendering any html. In fact, I don't even think presenters should render html... that's what views are for. However, if you need the ability to call this service to display html in different places, then I suppose this could be a good solution. Otherwise, I think a presenter/view combo would be better.

The problem with the html-service is that it's difficult to override. I can't define my own service and us it in place of this, I have to override the whole thing, and I might only need to make cosmetic changes, which suggests a view would be better. Our current pattern is to define presenters and services in the controller so one can override them and define their own there.

If you wanted to go with a presenter, then it's just a plain ruby class that encompasses a fixity status, whatever that is. If it's just a single record from the database table, then you might not need one and passing the AR object to the view is enough. If a "fixity status" is multiple records, or a single record with some logic anchored around it, then the presenter can take care of that and then the view receives the presenter object and renders its contents with the appropriate html tags, etc.